homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -528,8 +556,9 @@ async def timestamped_stream() -> AsyncIterable[tuple[bytes, int]]:
             timestamp_ms = 0
             async for chunk in stream:
                 if wav_writer is not None:
-                    _LOGGER.warning(chunk.replace(b" ", b"x"))
-                    wav_writer.writeframes(chunk)
+                    await self.hass.async_add_executor_job(
```

Comment:
This is very inefficient. It will do a lot of trips to the executor, potentially making it run slower than that the audio comes in. Can't we make a debug thread that we spawn, which listens to a queue and also allows for a specific value to be passed to the queue to delete it at the end?

Reply:
I'm using a thread now with a queue, which is cleaned up at the run end.